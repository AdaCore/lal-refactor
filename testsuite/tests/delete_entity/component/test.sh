cp ./main.orig ./main.adb
delete_entity -P default.gpr -S main.adb -L 2 -R 26
echo "--  (main.orig) -L 2 -R 26"
cat ./main.adb

cp ./main2.orig ./main.adb
delete_entity -P default.gpr -S main.adb -L 2 -R 23
echo "--  (main2.orig) -L 2 -R 23"
cat ./main.adb

cp ./main3.orig ./main.adb
delete_entity -P default.gpr -S main.adb -L 3 -R 7 
echo "--  (main3.orig) -L 3 -R 7"
cat ./main.adb
