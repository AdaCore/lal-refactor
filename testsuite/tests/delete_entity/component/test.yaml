title: delete_component

description: delete_enity tool test (component declaration)

driver: shell_script
