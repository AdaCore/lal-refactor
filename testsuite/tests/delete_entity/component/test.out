--  (main.orig) -L 2 -R 26
procedure Main is
   type Rec is record A, C : Integer; end record;

   for Rec use record
      A at 0 range 0 .. 31;
      C at 8 range 0 .. 31;
   end record;
   
   V : Rec :=
     (A => 1,
      C => 3);

begin
   null;
end Main;
--  (main2.orig) -L 2 -R 23
procedure Main is
   type Rec is record null; end record;


   V : Rec := (null record);

begin
   null;
end Main;
--  (main3.orig) -L 3 -R 7
procedure Main is
   type Rec is record

      B : Integer;
   end record;

   for Rec use record
      B at 4 range 0 .. 31;
   end record;
   
   V : Rec :=
     (B => 2);

begin
   null;
end Main;
